src/hooks/useResizer.js

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { addActions, actions } from "../actions";
+import { defaultState } from "./useTableState";
+
+defaultState.resizedColumns = {};
+
+addActions({
+  resizeColumn: "__resizeColumn__"
+});
+
+const propTypes = {
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      width: PropTypes.number
+    })
+  ),
+  onResizedChange: PropTypes.func
+};
+
+export const useResizer = props => {
+  PropTypes.checkPropTypes(propTypes, props, "property", "useResizer");
+
+  const {
+    columns,
+    hooks: { headers: headerHooks },
+    state: [{ resizedColumns }, setState],
+    onResizedChange
+  } = props;
+
+  const resizeColumn = (e, intialPosition, initialWidth, columnIndex) => {
+    const { clientX: currentPosition } = e;
+    const positionXDelta = currentPosition - intialPosition;
+    let updatedWidth = initialWidth + positionXDelta > 0 ? initialWidth + positionXDelta : 0;
+
+    if (columns[columnIndex].minWidth && updatedWidth < columns[columnIndex].minWidth) {
+      updatedWidth = columns[columnIndex].minWidth;
+    }
+
+    if (columns[columnIndex].maxWidth && updatedWidth > columns[columnIndex].maxWidth) {
+      updatedWidth = columns[columnIndex].maxWidth;
+    }
+
+    if (updatedWidth !== initialWidth) {
+      setState(old => {
+        let { resizedColumns } = old;
+
+        resizedColumns[columnIndex] = updatedWidth;
+        columns[columnIndex].width = updatedWidth;
+
+        if (onResizedChange) {
+          onResizedChange(resizedColumns, e);
```

Comment:
Triggering change functions is now an antipattern, since the user can now hoist the state out of react-table and use an effect to watch it, eg.

```
const state = useTableState()

const [{ resizedColumns }] = state

useEffect(() => {
  // resizedColumnsChanged
}, [resizedColumns])

useTable({
  ... 
  state
})

Reply:
Haha, I mean't to change that.  Since it worked I must have forgotten about it